### PR TITLE
Add holidays to full day events

### PR DESCRIPTION
**Pull request by itayf9**
I've removed duplications from full day events.
Also, I've fixed the set of holidays data type (instead of Set<String> it's now a List<Holiday>).
I've added holidays that don't have a date that matches any full day event, to the full day events list, in order to be shown to the user to decide.

### CODE REVIEW

**Comment by Danielsio** [comment]
why there are conflicts ?

**Review by Danielsio — APPROVED** [review]

**Comment by itayf9** [comment]
The conflict is not a real one. It ws caused by a line that was copied and pasted with more lines, and the git cannot thinks it was overwritten with other content.
Anyway, it will be resolved right away.

**Review by Danielsio — APPROVED** [review]

**Comment by Danielsio** [comment]
It's advised to resolve conflicts with rebase action instead of merge.
Main reason is because merge action is forcing a new commit in the history, which now gets in the PR and make it hard for maintainers to keep track of overall cahnges, and on the other hand rebase action when done correctly isnt creating a new commit and thus keeps the PR with single commit, conflict resolution is the same, I can share a paper one of my mentors wrote about this matter

File: src/main/java/com/example/planit/engine/CalendarEngine.java
Comment (oshriFadlon1) [inline src/main/java/com/example/planit/engine/CalendarEngine.java:1]: if we not need the comment 1424 - 1442.
we can remove it.

Comment (itayf9) [reply src/main/java/com/example/planit/engine/CalendarEngine.java:1]: removed.

**Pull request merged**